### PR TITLE
Refactor references and schema mapping

### PR DESCRIPTION
**Pull request by marikaner**
Refactoring of the document referenced and schemarefmapping into one object.

### CODE REVIEW

File: packages/openapi-generator/src/parser/refs.ts [hunk 1]
```diff
@@ -15,3 +19,81 @@ export function resolveObject<T>(
 ): T {
   return isReferenceObject(obj) ? refs.get(obj.$ref) : obj;
 }
+
+export async function createRefs(
```
Comment (FrankEssenberger) [inline packages/openapi-generator/src/parser/refs.ts:23]: some docs for the methods

Comment (FrankEssenberger) [reply packages/openapi-generator/src/parser/refs.ts:23]: I would move this as a static method of the class -> builder

File: packages/openapi-generator/src/parser/refs.ts [hunk 2]
```diff
@@ -15,3 +19,81 @@ export function resolveObject<T>(
...
+    if (!schemaNaming) {
+      throw new Error(
+        `Could not find schema naming for reference path '${ref}'. Schema does not exist.`
+      );
+    }
+    return schemaNaming;
+  }
+}
+
+function parseSchemaRefMapping(document: OpenAPIV3.Document): SchemaRefMapping {
```
Comment (FrankEssenberger) [inline packages/openapi-generator/src/parser/refs.ts:70]: Also matter of taste, I would put this as a private method in the class and call this in the constructor.

**Review by FrankEssenberger — APPROVED** [review]
LGTM - I like your class. Consider my comments on the methods and where to put them.

File: packages/openapi-generator/src/parser/refs.ts (hunk 1 above)
Comment (marikaner) [reply packages/openapi-generator/src/parser/refs.ts:23]: Done, but I kept an additional function for convenience as this much shorter and "functional" to write.

File: packages/openapi-generator/src/parser/refs.ts (hunk 2 above)
Comment (marikaner) [reply packages/openapi-generator/src/parser/refs.ts:70]: done

**Pull request merged**